src/msg/async/dpdk/DPDK.cc

```diff
@@ -642,6 +650,119 @@ DPDKQueuePair::DPDKQueuePair(CephContext *c, EventCenter *cen, DPDKDevice* dev,
+                nic_stats_border,_port_idx,nic_stats_border);
+  bl.append(buf,len);
+  len = sprintf(buf,"RX-packets:%-10" PRIu64 " qp-packets:%-10" PRIu64
+		"RX-missed: %-10" PRIu64 " RX-bytes: %-" PRIu64 "\n",
+		stats.ipackets,rx_fragments,stats.imissed,stats.ibytes);
+  bl.append(buf,len);
+  len=sprintf(buf," RX-errors: %-" PRIu64 "\n",stats.ierrors);
+  bl.append(buf,len);
+  len=sprintf(buf," RX-nombuf: %-10" PRIu64 " tx free mbuf: %-10" PRIu64
+              " rx free mbuf: %-10" PRIu64 "\n",stats.rx_nombuf,tx_free_cnt,
+              rx_free_cnt);
+  bl.append(buf,len);
+  len=sprintf(buf," TX-packets: %-10" PRIu64 " qp-packets: %-10" PRIu64
+              " TX-errors: %-10" PRIu64 "TX-bytes: %-" PRIu64 "\n",
+              stats.opackets,tx_fragments,stats.oerrors,stats.obytes);
+  bl.append(buf,len);  
+  diff_cycles = prev_cycles[_port_idx];
+  prev_cycles[_port_idx] = rte_rdtsc();
+  if (diff_cycles > 0)
+    diff_cycles = prev_cycles[_port_idx] - diff_cycles;
+
+  diff_pkts_rx = (stats.ipackets > prev_pkts_rx[_port_idx]) ?
+	         (stats.ipackets - prev_pkts_rx[_port_idx]) : 0;
+  diff_pkts_tx = (stats.opackets > prev_pkts_tx[_port_idx]) ?
+	         (stats.opackets - prev_pkts_tx[_port_idx]) : 0;
+  prev_pkts_rx[_port_idx] = stats.ipackets;
+  prev_pkts_tx[_port_idx] = stats.opackets;
+  mpps_rx = diff_cycles > 0 ? diff_pkts_rx * rte_get_tsc_hz()/diff_cycles : 0;
+  mpps_tx = diff_cycles > 0 ? diff_pkts_tx * rte_get_tsc_hz()/diff_cycles : 0;
+  len = sprintf(buf, "\n Throughput (since last show)\n");
+  bl.append(buf,len);
+  len = sprintf(buf," Rx-pps: %12" PRIu64 "\n Tx-pps: %12" PRIu64"\n",
+	        mpps_rx,mpps_tx);
+  bl.append(buf,len);
+
+  len = sprintf(buf," %s############################%s\n",
+		nic_stats_border,nic_stats_border);
+  bl.append(buf,len);
+}
+
+void DPDKDevice::nic_xstats_dump(bufferlist &bl)
+{
+  struct rte_eth_xstat_name *xstats_names;
+  struct rte_eth_xstat *xstats;
+  int cnt_xstats;
+  char buf[LINE_MAX];
+  int len;
+
+  //Get count
+  cnt_xstats = rte_eth_xstats_get_names(_port_idx,NULL,0); 
+  if (cnt_xstats < 0) {
+    ldout(cct,0) << "Error: Cannot get count of xstats" << dendl;
+    return ;
+  }
+ 
+  //Get id-name lookup table
+  xstats_names = (struct rte_eth_xstat_name *) malloc(sizeof(struct rte_eth_xstat_name) * cnt_xstats);
+  if (xstats_names == nullptr) {
+    return;
+  }
+  if (cnt_xstats != rte_eth_xstats_get_names(_port_idx,xstats_names,cnt_xstats)) {
+    ldout(cct,0) << "Error: Cannot get xstats lookup" << dendl;
+    free(xstats_names);
+    return;
+  }
+
+  //Get stats themselves
+  xstats = (struct rte_eth_xstat *)malloc(sizeof(struct rte_eth_xstat) * cnt_xstats);
+  if (xstats == nullptr) {
+    ldout(cct,0) << "Cannot allocate memory for xstats" << dendl;
+    free(xstats_names);
+    return;
+  }
+  if (cnt_xstats != rte_eth_xstats_get(_port_idx, xstats, cnt_xstats)) {
+    ldout(cct,0)<< "Error: Unable to get xstats" << dendl;
```

Comment:
again, please add a space after `,`. also, i'd suggest use `ldout(cct, 1)` for this error.